### PR TITLE
Reader Tags Reduxification (4): data-layer for follow tag

### PR DESCRIPTION
**Pull request by samouri**
### Table of Contents
1. new util for /read/tags data-layer: https://github.com/Automattic/wp-calypso/pull/11733
2. updating read/tags data-layer: https://github.com/Automattic/wp-calypso/pull/11627
3. unfollowing data-layer:  https://github.com/Automattic/wp-calypso/pull/11647
4. **following data-layer: https://github.com/Automattic/wp-calypso/pull/11649**
5. reducers + actions: https://github.com/Automattic/wp-calypso/pull/11734
6. query components + selectors: https://github.com/Automattic/wp-calypso/pull/11736
7. connecting: https://github.com/Automattic/wp-calypso/pull/11737
8. deleting old code: https://github.com/Automattic/wp-calypso/pull/11738

### Summary
This PR introduces the data-layer for the follow tag api.
If an error occurs while attempting to follow a tag it creates an error notice (can test by dispatching in browser).

### CODE REVIEW

File: client/state/data-layer/wpcom/read/tags/mine/new/index.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { fromApi } from 'state/data-layer/wpcom/read/tags/utils';
+import { errorNotice } from 'state/notices/actions';
+import { translate } from 'i18n-calypso';
+
+export function requestFollowTag( store, action, next ) {
+	store.dispatch( http( {
+		path: `/read/tags/${ action.payload.slug }/mine/new`,
+		method: 'POST',
+		apiVersion: '1.0',
```
Comment (blowery) [inline client/state/data-layer/wpcom/read/tags/mine/new/index.js]: are we using 1.0 currently? most stuff should default to 1.1.

Comment (samouri) [inline client/state/data-layer/wpcom/read/tags/mine/new/index.js]: It didn't work with 1.2, I don't think I tried 1.1, I'll test it out

File: client/state/data-layer/wpcom/read/tags/mine/new/index.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		isFollowing: true,
+	};
+
+	store.dispatch( receiveTagsAction( {
+		payload: [ followedTag ],
+	} ) );
+}
+
+export function receiveError( store, action, next, error ) {
+	const errorText = translate( 'Could not unfollow tag: %(tag)', { tag: action.payload.slug } );
```
Comment (blowery) [inline client/state/data-layer/wpcom/read/tags/mine/new/index.js]: should be follow tag

File: client/state/data-layer/wpcom/read/tags/mine/new/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	store.dispatch( errorNotice( 'Could not follow tag' ) );
```
Comment (blowery) [inline client/state/data-layer/wpcom/read/tags/mine/new/index.js]: should be translated, likely using the variable just above

File: client/state/data-layer/wpcom/read/tags/mine/new/index.js (hunk 1 above)
Comment (samouri) [inline client/state/data-layer/wpcom/read/tags/mine/new/index.js]: 1.1 works! i'll change both follow/unfollow prs

File: client/state/data-layer/wpcom/read/tags/mine/new/index.js (hunk 3 above)
Comment (samouri) [inline client/state/data-layer/wpcom/read/tags/mine/new/index.js]: oops, ill fix

**Pull request merged**